diff -urN openssl-3.5.4-orig/Configurations/windows-makefile.tmpl openssl-3.5.4/Configurations/windows-makefile.tmpl
--- openssl-3.5.4-orig/Configurations/windows-makefile.tmpl	2025-09-30 20:37:40.000000000 +0800
+++ openssl-3.5.4/Configurations/windows-makefile.tmpl	2025-12-11 11:49:55.585649500 +0800
@@ -155,6 +155,11 @@
                          && $unified_info{attributes}->{scripts}->{$_}->{misc} }
                   @{$unified_info{scripts}})
 -}
+INSTALL_EXPORTERS_PKGCONFIG={-
+        join(" ", map { quotify1($_) }
+                  grep { $unified_info{attributes}->{generate}->{$_}->{exporter} eq 'pkg-config'}
+                  sort keys %{$unified_info{generate}})
+-}
 INSTALL_EXPORTERS_CMAKE={-
         join(" ", map { quotify1($_) }
                   grep { $unified_info{attributes}->{generate}->{$_}->{exporter} eq 'cmake'}
@@ -254,6 +259,7 @@
 libdir={- file_name_is_absolute($libdir)
           ? $libdir : '$(INSTALLTOP)\$(LIBDIR)' -}
 
+PKGCONFIGDIR=$(libdir)\pkgconfig
 CMAKECONFIGDIR=$(libdir)\cmake\OpenSSL
 
 ##### User defined commands and flags ################################
@@ -566,6 +572,8 @@
 	@"$(PERL)" "$(SRCDIR)\util\copy.pl" $(INSTALL_LIBS) "$(libdir)"
 	@if "$(SHLIBS)"=="" \
 	 "$(PERL)" "$(SRCDIR)\util\copy.pl" ossl_static.pdb "$(libdir)"
+	@"$(PERL)" "$(SRCDIR)\util\mkdir-p.pl" "$(PKGCONFIGDIR)"
+	@"$(PERL)" "$(SRCDIR)\util\copy.pl" $(INSTALL_EXPORTERS_PKGCONFIG) "$(PKGCONFIGDIR)"
 	@"$(PERL)" "$(SRCDIR)\util\mkdir-p.pl" "$(CMAKECONFIGDIR)"
 	@"$(PERL)" "$(SRCDIR)\util\copy.pl" $(INSTALL_EXPORTERS_CMAKE) "$(CMAKECONFIGDIR)"
 
diff -urN openssl-3.5.4-orig/exporters/pkg-config/libcrypto.pc.in openssl-3.5.4/exporters/pkg-config/libcrypto.pc.in
--- openssl-3.5.4-orig/exporters/pkg-config/libcrypto.pc.in	2025-09-30 20:37:40.000000000 +0800
+++ openssl-3.5.4/exporters/pkg-config/libcrypto.pc.in	2025-12-11 11:52:22.186791700 +0800
@@ -1,4 +1,23 @@
-prefix={- $OpenSSL::safe::installdata::PREFIX[0] -}
+{-
+   # Unixifying of paths, because that's what CMake understands.
+   # This assumes that the given path is absolute.
+   sub unixify {
+       use File::Spec;
+       use File::Spec::Unix;
+
+       my $path = shift;
+       my $no_file = shift;
+       (my $volume, $directories, $file) = File::Spec->splitpath($path, $no_file // ());
+       $directories = File::Spec::Unix->catdir(File::Spec->splitdir($directories));
+
+       # We know that $volume is ignored by File::Spec::Unix, so we handle it specially
+       # according to the conventions of the platform.
+       $volume =~ s|^(.*?):$|/$1| if $^O eq "vms"; # On VMS, DEV: -> /DEV
+
+       return $volume . File::Spec::Unix->catpath('', $directories, $no_file ? () : $file);
+   }
+-}
+prefix={- unixify($OpenSSL::safe::installdata::PREFIX[0], 1); -}
 exec_prefix=${prefix}
 libdir={- if (defined $OpenSSL::safe::installdata::LIBDIR_REL_PREFIX[0]) {
               my $x = '';
@@ -17,7 +36,7 @@
 Name: OpenSSL-libcrypto
 Description: OpenSSL cryptography library
 Version: {- $OpenSSL::safe::installdata::VERSION -}
-Libs: -L${libdir} -lcrypto
+Libs: -L${libdir} -llibcrypto
 Libs.private: {- join(' ', @OpenSSL::safe::installdata::LDLIBS) -}
 Cflags:{- $OUT = ' -I${includedir}';
           if (scalar @OpenSSL::safe::installdata::INCLUDEDIR_REL_PREFIX > 1) {
diff -urN openssl-3.5.4-orig/exporters/pkg-config/libssl.pc.in openssl-3.5.4/exporters/pkg-config/libssl.pc.in
--- openssl-3.5.4-orig/exporters/pkg-config/libssl.pc.in	2025-09-30 20:37:40.000000000 +0800
+++ openssl-3.5.4/exporters/pkg-config/libssl.pc.in	2025-12-11 11:53:07.685120600 +0800
@@ -1,4 +1,23 @@
-prefix={- $OpenSSL::safe::installdata::PREFIX[0] -}
+{-
+   # Unixifying of paths, because that's what CMake understands.
+   # This assumes that the given path is absolute.
+   sub unixify {
+       use File::Spec;
+       use File::Spec::Unix;
+
+       my $path = shift;
+       my $no_file = shift;
+       (my $volume, $directories, $file) = File::Spec->splitpath($path, $no_file // ());
+       $directories = File::Spec::Unix->catdir(File::Spec->splitdir($directories));
+
+       # We know that $volume is ignored by File::Spec::Unix, so we handle it specially
+       # according to the conventions of the platform.
+       $volume =~ s|^(.*?):$|/$1| if $^O eq "vms"; # On VMS, DEV: -> /DEV
+
+       return $volume . File::Spec::Unix->catpath('', $directories, $no_file ? () : $file);
+   }
+-}
+prefix={- unixify($OpenSSL::safe::installdata::PREFIX[0], 1); -}
 exec_prefix=${prefix}
 libdir={- if (defined $OpenSSL::safe::installdata::LIBDIR_REL_PREFIX[0]) {
               my $x = '';
@@ -16,7 +35,7 @@
 Description: Secure Sockets Layer and cryptography libraries
 Version: {- $OpenSSL::safe::installdata::VERSION -}
 Requires.private: libcrypto
-Libs: -L${libdir} -lssl
+Libs: -L${libdir} -llibssl
 Cflags:{- $OUT = ' -I${includedir}';
           if (scalar @OpenSSL::safe::installdata::INCLUDEDIR_REL_PREFIX > 1) {
               $OUT = '';
diff -urN openssl-3.5.4-orig/exporters/pkg-config/openssl.pc.in openssl-3.5.4/exporters/pkg-config/openssl.pc.in
--- openssl-3.5.4-orig/exporters/pkg-config/openssl.pc.in	2025-09-30 20:37:40.000000000 +0800
+++ openssl-3.5.4/exporters/pkg-config/openssl.pc.in	2025-12-11 11:53:36.178480000 +0800
@@ -1,4 +1,23 @@
-prefix={- $OpenSSL::safe::installdata::PREFIX[0] -}
+{-
+   # Unixifying of paths, because that's what CMake understands.
+   # This assumes that the given path is absolute.
+   sub unixify {
+       use File::Spec;
+       use File::Spec::Unix;
+
+       my $path = shift;
+       my $no_file = shift;
+       (my $volume, $directories, $file) = File::Spec->splitpath($path, $no_file // ());
+       $directories = File::Spec::Unix->catdir(File::Spec->splitdir($directories));
+
+       # We know that $volume is ignored by File::Spec::Unix, so we handle it specially
+       # according to the conventions of the platform.
+       $volume =~ s|^(.*?):$|/$1| if $^O eq "vms"; # On VMS, DEV: -> /DEV
+
+       return $volume . File::Spec::Unix->catpath('', $directories, $no_file ? () : $file);
+   }
+-}
+prefix={- unixify($OpenSSL::safe::installdata::PREFIX[0], 1); -}
 exec_prefix=${prefix}
 libdir={- if (defined $OpenSSL::safe::installdata::LIBDIR_REL_PREFIX[0]) {
               my $x = '';
