diff -urN libvideogfx-1.0.8-orig/libvideogfx/graphics/fileio/png.cc libvideogfx-1.0.8/libvideogfx/graphics/fileio/png.cc
--- libvideogfx-1.0.8-orig/libvideogfx/graphics/fileio/png.cc	2013-06-24 20:55:00.000000000 +0800
+++ libvideogfx-1.0.8/libvideogfx/graphics/fileio/png.cc	2025-11-16 03:04:07.223848300 +0800
@@ -87,8 +87,11 @@
 
   void ReadImage_PNG(Image<Pixel>& img, istream& is)
   {
+#ifdef _MSC_VER
+    assert(is);
+#else
     assert(is != NULL); // , "Open stream first.");
-
+#endif
     png_structp png_ptr;
     png_infop info_ptr;
     png_uint_32 width, height;
diff -urN libvideogfx-1.0.8-orig/libvideogfx/graphics/fileio/unified_reader.cc libvideogfx-1.0.8/libvideogfx/graphics/fileio/unified_reader.cc
--- libvideogfx-1.0.8-orig/libvideogfx/graphics/fileio/unified_reader.cc	2013-06-24 20:55:00.000000000 +0800
+++ libvideogfx-1.0.8/libvideogfx/graphics/fileio/unified_reader.cc	2025-11-20 08:57:26.102443200 +0800
@@ -488,7 +488,7 @@
     int  AskNFrames() const { return INT_MAX; }
     bool IsEOF() const { return false; }
 
-    bool SkipToImage(int nr) { id = (nr%3); }
+    bool SkipToImage(int nr) { id = (nr%3); return false; }
     void ReadImage(Image<Pixel>& img)
     {
       ImageParam spec(d_w,d_h);
diff -urN libvideogfx-1.0.8-orig/libvideogfx/types.hh libvideogfx-1.0.8/libvideogfx/types.hh
--- libvideogfx-1.0.8-orig/libvideogfx/types.hh	2013-06-24 20:45:17.000000000 +0800
+++ libvideogfx-1.0.8/libvideogfx/types.hh	2025-11-20 08:54:35.255120400 +0800
@@ -35,7 +35,7 @@
 #ifndef LIBVIDEOGFX_TYPES_HH
 #define LIBVIDEOGFX_TYPES_HH
 
-#if defined(__GCC__) && (__GNUC__ < 4) || (__GNUC__ == 4 && __GNUC_MINOR__ < 5)
+#if defined(__GCC__) && (__GNUC__ < 4) || (__GNUC__ == 4 && __GNUC_MINOR__ < 5) || defined(_MSC_VER)
 #include <stdint.h>
 #else
 #include <cstdint>
@@ -116,4 +116,12 @@
 #define NULL 0
 #endif
 
+#ifdef _WIN64
+typedef __int64 ssize_t;
+typedef __int64 pid_t;
+#else
+typedef int ssize_t;
+typedef int     pid_t;
+#endif
+
 #endif
diff -urN libvideogfx-1.0.8-orig/libvideogfx/utility/bitstream/bitbuffer.cc libvideogfx-1.0.8/libvideogfx/utility/bitstream/bitbuffer.cc
--- libvideogfx-1.0.8-orig/libvideogfx/utility/bitstream/bitbuffer.cc	2013-06-24 20:55:00.000000000 +0800
+++ libvideogfx-1.0.8/libvideogfx/utility/bitstream/bitbuffer.cc	2025-11-16 03:04:07.223848300 +0800
@@ -58,7 +58,7 @@
 
   void BitBuffer::WriteBitsMasked(uint32 bits,int nBits)
   {
-    uint32 long mask=1;
+    uint32 mask=1;
     mask<<=nBits;
     mask--;
 
