diff -urN Vol-1.5.4-orig/Vol/configure.ac Vol-1.5.4/Vol/configure.ac
--- Vol-1.5.4-orig/Vol/configure.ac	2019-02-23 05:42:17.000000000 +0800
+++ Vol-1.5.4/Vol/configure.ac	2026-01-13 18:22:59.274571300 +0800
@@ -20,10 +20,6 @@
 This file is part of the open source package Coin which is distributed
 under the Eclipse Public License.])
 
-# List one file in the package so that the configure script can test
-# whether the package is actually there
-AC_CONFIG_SRCDIR(src/VolVolume.cpp)
-
 # Where should everything be installed by default?  Here, we want it
 # to be installed directly in 'bin', 'lib', 'include' subdirectories
 # of the directory where configure is run.  The default would be
@@ -34,32 +30,23 @@
 #                         Standard build tool stuff                         #
 #############################################################################
 
-# Get the system type
-AC_CANONICAL_BUILD
-
-# If this project depends on external projects, the Externals file in
-# the source root directory contains definition of where to find those
-# externals.  The following macro ensures that those externals are
-# retrieved by svn if they are not there yet.
-AC_COIN_PROJECTDIR_INIT(Vol,6:4:5)
-
-# Check if user wants to produce debugging code
-AC_COIN_DEBUG_COMPILE(Vol)
+AC_COIN_INITIALIZE
 
 # Get the name of the C++ compiler and appropriate compiler options
 AC_COIN_PROG_CXX
 
 # Initialize automake and libtool
-AC_COIN_INIT_AUTO_TOOLS
+AC_COIN_PROG_LIBTOOL
 
 #############################################################################
 #                              COIN components                              #
 #############################################################################
 
-AC_COIN_CHECK_PACKAGE(CoinUtils, [coinutils],     [VolLib])
-AC_COIN_CHECK_PACKAGE(Osi,       [osi],           [OsiVolLib])
-AC_COIN_CHECK_PACKAGE(OsiTests,  [osi-unittests])
-AC_COIN_CHECK_PACKAGE(Sample,    [coindatasample])
+AC_COIN_CHK_PKG(CoinUtils,[CoinUtils VolLib],[coinutils])
+AC_COIN_CHK_PKG(Osi,[OsiVolLib],[osi])
+AC_COIN_CHK_PKG(OsiTests,[OsiVolUnitTest],[osi-unittests])
+AC_COIN_CHK_PKG(Sample,,[coindatasample],,dataonly)
+AC_COIN_FINALIZE_FLAGS([CoinUtils VolLib OsiVolLib])
 
 ##############################################################################
 #                   VPATH links for example input files                      #
@@ -100,7 +87,7 @@
 # even though we do not use the config_vol.h file, we config a header file here
 # this is for the compilation of OsiVol, which works much better with -DHAVE_CONFIG_H
 # instead of a list of single defines (-DPACKAGE_NAME=...) in the compiler flags
-AC_CONFIG_HEADER([inc/config_vol.h])
+AC_CONFIG_HEADERS([inc/config_vol.h])
 
 # Finally, we let configure write all the output...
 AC_COIN_FINALIZE
diff -urN Vol-1.5.4-orig/Vol/Makefile.am Vol-1.5.4/Vol/Makefile.am
--- Vol-1.5.4-orig/Vol/Makefile.am	2019-02-23 05:42:17.000000000 +0800
+++ Vol-1.5.4/Vol/Makefile.am	2026-01-13 18:22:59.277536600 +0800
@@ -83,7 +83,6 @@
 #                  Installation of the addlibs file                    #
 ########################################################################
 
-pkgconfiglibdir = $(libdir)/pkgconfig
 pkgconfiglib_DATA = vol.pc
 
 if COIN_HAS_OSI
@@ -98,12 +97,8 @@
 	PKG_CONFIG_PATH=@COIN_PKG_CONFIG_PATH@:$(DESTDIR)$(pkgconfiglibdir) \
 	$(PKG_CONFIG) --libs vol > $(addlibsdir)/vol_addlibs.txt
 else
-if COIN_CXX_IS_CL
-	echo "-libpath:`$(CYGPATH_W) @abs_lib_dir@` libVol.lib @VOLLIB_LIBS_INSTALLED@" > $(addlibsdir)/vol_addlibs.txt
-else
 	echo -L@abs_lib_dir@ -lVol @VOLLIB_LIBS_INSTALLED@ > $(addlibsdir)/vol_addlibs.txt
 endif
-endif
 
 uninstall-hook:
 	rm -f $(addlibsdir)/vol_addlibs.txt
diff -urN Vol-1.5.4-orig/Vol/src/Makefile.am Vol-1.5.4/Vol/src/Makefile.am
--- Vol-1.5.4-orig/Vol/src/Makefile.am	2019-02-23 05:42:17.000000000 +0800
+++ Vol-1.5.4/Vol/src/Makefile.am	2026-01-13 18:22:59.279531500 +0800
@@ -31,16 +31,13 @@
 # "cygpath" stuff is necessary to compile with native compilers on Windows.
 # "top_srcdir" refers to the basic directory for the main package that is
 # being compiled.
-AM_CPPFLAGS = ${COINUTILS_CFLAGS}
-
-# This line is necessary to allow VPATH compilation
-DEFAULT_INCLUDES = -I. -I`$(CYGPATH_W) $(srcdir)` -I$(top_builddir)/inc
+AM_CPPFLAGS = ${COINUTILS_CFLAGS} -I. -I`$(CYGPATH_W) $(srcdir)` -I$(top_builddir)/inc
 
 ########################################################################
 #                Headers that need to be installed                     #
 ########################################################################
 
 # Here list all the header files that are required by a user of the library,
-# and that therefore should be installed in 'include/coin'
-includecoindir = $(includedir)/coin
+# and that therefore should be installed in 'include/coin-or'
+includecoindir = $(includedir)/coin-or
 includecoin_HEADERS = VolVolume.hpp
diff -urN Vol-1.5.4-orig/Vol/src/OsiVol/Makefile.am Vol-1.5.4/Vol/src/OsiVol/Makefile.am
--- Vol-1.5.4-orig/Vol/src/OsiVol/Makefile.am	2019-02-23 05:42:17.000000000 +0800
+++ Vol-1.5.4/Vol/src/OsiVol/Makefile.am	2026-01-15 13:55:10.347987400 +0800
@@ -21,19 +21,14 @@
 	OsiVolSolverInterfaceIO.cpp
 
 # List all additionally required libraries
-if DEPENDENCY_LINKING
-libOsiVol_la_LIBADD = $(OSIVOLLIB_LIBS) ../libVol.la
-endif
+libOsiVol_la_LIBADD = $(OSIVOLLIB_LFLAGS) ../libVol.la
 
 # This is for libtool (on Windows)
 libOsiVol_la_LDFLAGS = $(LT_LDFLAGS)
 
 # Here list all include flags, relative to this "srcdir" directory.  This
 # "cygpath" stuff is necessary to compile with native compilers on Windows.
-AM_CPPFLAGS = -I`$(CYGPATH_W) $(srcdir)/..` ${COINUTILS_CFLAGS} ${OSI_CFLAGS}
-
-# This line is necessary to allow VPATH compilation
-DEFAULT_INCLUDES = -I. -I`$(CYGPATH_W) $(srcdir)` -I$(top_builddir)/inc
+AM_CPPFLAGS = -I`$(CYGPATH_W) $(srcdir)/..` ${COINUTILS_CFLAGS} ${OSIVOLLIB_CFLAGS} -I. -I`$(CYGPATH_W) $(srcdir)` -I$(top_builddir)/inc
 
 ########################################################################
 #                Headers that need to be installed                     #
@@ -41,5 +36,5 @@
 
 # Here list all the header files that are required by a user of the library,
 # and that therefore should be installed in 'includedir/coin'
-includecoindir = $(includedir)/coin
+includecoindir = $(includedir)/coin-or
 includecoin_HEADERS = OsiVolSolverInterface.hpp
diff -urN Vol-1.5.4-orig/Vol/src/OsiVol/osi-vol.pc.in Vol-1.5.4/Vol/src/OsiVol/osi-vol.pc.in
--- Vol-1.5.4-orig/Vol/src/OsiVol/osi-vol.pc.in	2019-02-23 05:42:17.000000000 +0800
+++ Vol-1.5.4/Vol/src/OsiVol/osi-vol.pc.in	2026-01-13 18:22:59.283520200 +0800
@@ -1,12 +1,12 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@/coin
+includedir=@includedir@/coin-or
 
 Name: OsiVol
 Description: COIN-OR Open Solver Interface for Vol
 URL: https://projects.coin-or.org/Osi
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lOsiVol @OSIVOLLIB_PCLIBS@
+Libs: -L${libdir} -lOsiVol @OSIVOLLIB_LFLAGS@
 Cflags: -I${includedir}
-Requires: vol @OSIVOLLIB_PCREQUIRES@
+Requires: vol @OSIVOLLIB_PCFILES@
diff -urN Vol-1.5.4-orig/Vol/src/OsiVol/osi-vol-uninstalled.pc.in Vol-1.5.4/Vol/src/OsiVol/osi-vol-uninstalled.pc.in
--- Vol-1.5.4-orig/Vol/src/OsiVol/osi-vol-uninstalled.pc.in	2019-02-23 05:42:17.000000000 +0800
+++ Vol-1.5.4/Vol/src/OsiVol/osi-vol-uninstalled.pc.in	2026-01-13 18:22:59.284517600 +0800
@@ -5,6 +5,6 @@
 Description: COIN-OR Open Solver Interface for Vol
 URL: https://projects.coin-or.org/Osi
 Version: @PACKAGE_VERSION@
-Libs: ${libdir}/libOsiVol.la @OSIVOLLIB_PCLIBS@
+Libs: ${libdir}/libOsiVol.la @OSIVOLLIB_LFLAGS@
 Cflags: -I@abs_source_dir@/src/OsiVol
-Requires: vol @OSIVOLLIB_PCREQUIRES@
+Requires: vol @OSIVOLLIB_PCFILES@
diff -urN Vol-1.5.4-orig/Vol/vol.pc.in Vol-1.5.4/Vol/vol.pc.in
--- Vol-1.5.4-orig/Vol/vol.pc.in	2019-02-23 05:42:17.000000000 +0800
+++ Vol-1.5.4/Vol/vol.pc.in	2026-01-13 18:22:59.285514500 +0800
@@ -1,12 +1,12 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@/coin
+includedir=@includedir@/coin-or
 
 Name: Vol
 Description: Volume Algorithm
 URL: https://projects.coin-or.org/Vol
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lVol @VOLLIB_PCLIBS@
+Libs: -L${libdir} -lVol @VOLLIB_LFLAGS@
 Cflags: -I${includedir}
-Requires: @VOLLIB_PCREQUIRES@
+Requires: @VOLLIB_PCFILES@
diff -urN Vol-1.5.4-orig/Vol/vol-uninstalled.pc.in Vol-1.5.4/Vol/vol-uninstalled.pc.in
--- Vol-1.5.4-orig/Vol/vol-uninstalled.pc.in	2019-02-23 05:42:17.000000000 +0800
+++ Vol-1.5.4/Vol/vol-uninstalled.pc.in	2026-01-13 18:22:59.286511900 +0800
@@ -5,6 +5,6 @@
 Description: Volume Algorithm
 URL: https://projects.coin-or.org/Vol
 Version: @PACKAGE_VERSION@
-Libs: ${libdir}/libVol.la @VOLLIB_PCLIBS@
+Libs: ${libdir}/libVol.la @VOLLIB_LFLAGS@
 Cflags: -I@abs_source_dir@/src -I@ABSBUILDDIR@/inc
-Requires: @VOLLIB_PCREQUIRES@
+Requires: @VOLLIB_PCFILES@
